### PR TITLE
remove intersect from event storage

### PR DESCRIPTION
**Pull request by johannkm**
Mysql 8.0.35 has a change that breaks our usage of `intersect`. We primarily added this for run tags, but no longer use it there. Let's remove it from event logs too.

I don't have the full context here. @prha et al. could you give this some extra scrutiny

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17492** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17492" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/17492?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py [hunk 1]
```diff
@@ -237,10 +237,6 @@ def _add_cursor_limit_to_query(
 
         return query
 
-    @property
```
Comment (johannkm) [inline python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py:240]: unused

File: python_modules/libraries/dagster-mysql/dagster_mysql_tests/test_event_log.py [hunk 2]
```diff
@@ -116,6 +116,3 @@ def test_load_from_config(self, conn_string):
                 from_explicit = explicit_instance._event_storage  # noqa: SLF001
 
                 assert from_url.mysql_url == from_explicit.mysql_url
-
-    def test_materialization_tag_on_wipe(self, storage, instance):
```
Comment (johannkm) [inline python_modules/libraries/dagster-mysql/dagster_mysql_tests/test_event_log.py:120]: reverts https://github.com/dagster-io/dagster/pull/17482

File: python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py [hunk 3]
```diff
@@ -952,11 +935,7 @@ def _get_event_records(
         else:
             asset_details = None
 
-        if (
-            event_records_filter.tags
-            and not self.supports_intersect
-            and self.has_table(AssetEventTagsTable.name)
-        ):
+        if event_records_filter.tags and self.has_table(AssetEventTagsTable.name):
```
Comment (johannkm) [inline python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py:938]: i'm a little confused about this block- we were checking for intersect support but not using it?

File: python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py [hunk 4]
```diff
@@ -878,23 +878,6 @@ def _apply_filter_to_query(
                 isinstance(event_records_filter.asset_key, AssetKey),
                 "Asset key must be set in event records filter to filter by tags.",
             )
-            if self.supports_intersect:
```
Comment (jmsanders) [inline python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py:881]: Am I understanding correctly that the caller of this always filters after the fact...sometimes on a list that's already filtered, sometimes on one that isn't?

Otherwise, how could `self.supports_intersect` and `not self.supports_intersect` return the same results?

File: python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py (hunk 3 above)
Comment (johannkm) [reply python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py:938]: oh bc it uses _apply_filter_to_query after

File: python_modules/libraries/dagster-mysql/dagster_mysql/event_log/event_log.py [hunk 5]
```diff
@@ -208,12 +205,6 @@ def watch(self, run_id: str, cursor: Optional[str], callback: EventHandlerFn) ->
     def end_watch(self, run_id: str, handler: EventHandlerFn) -> None:
         self._event_watcher.unwatch_run(run_id, handler)
 
-    @property
-    def supports_intersect(self) -> bool:
-        return parse_mysql_version(self._mysql_version) >= parse_mysql_version(  # type: ignore  # (possible none)
-            MINIMUM_MYSQL_INTERSECT_VERSION
```
Comment (jmsanders) [inline python_modules/libraries/dagster-mysql/dagster_mysql/event_log/event_log.py:214]: Another option here is to change this to the version range that we know works 🤷 

But I'm not aware of any performance benchmarking we've done to show `INTERSECT` is better when available - probably not worth maintaining the complexity.

**Review by jmsanders — COMMENTED** [review]
Were the pinned version tests specifically to support this or are there other regressions we've caught with them? If the former, could we remove them?

File: python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py (hunk 4 above)
Comment (johannkm) [reply python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py:881]: yeah it's hard to follow. _apply_filter_to_query is only called by _get_event_records, which also checks supports_intersect and if it's not supported, does the filtering via _apply_tags_table_joins

**Comment by prha** [comment]
I think the pinned versions were just to support this, so we could rip out the `mysql-db-backcompat` fixtures from the docker file, etc.

@johannkm you can throw that over to me if you want

**Review by jmsanders — APPROVED** [review]
